Plugins/Creature/Creature.cpp

```diff
@@ -1793,23 +1793,24 @@ static uint8_t CNWSCreatureStats__GetClassLevel(CNWSCreatureStats* pThis, uint8_
 {
     auto retVal = s_GetClassLevelHook->CallOriginal<uint8_t>(pThis, nMultiClass, bUseNegativeLevel);
 
-    if (s_bAdjustCasterLevel || nMultiClass < pThis->m_nNumMultiClasses)
+    if (s_bAdjustCasterLevel && nMultiClass < pThis->m_nNumMultiClasses)
     {
         auto nClass = pThis->m_ClassInfo[nMultiClass].m_nClass;
 
         if (nClass != Constants::ClassType::Invalid)
         {
             int32_t nModifier = 0;
+            auto nLevelModifier = pThis->m_pBaseCreature->nwnxGet<int>("CASTERLEVEL_MODIFIER" + std::to_string(nClass));
```

Comment:
This should be after CASTERLEVEL_OVERRIDE so the override can exit early and skip reading the modifier.